### PR TITLE
[Feature] Support more csv paramters for broker load

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This PR supports more CSV attributes for broker load, just like stream load. Please reference to https://github.com/StarRocks/starrocks/pull/13582. 
For example, you can perform the CSV import with the following command for broker load.
```
LOAD LABEL test.labelname
(
    DATA INFILE("hdfs://hdfs_ip:hdfs_port/user/starrocks/csvfile.csv")
    INTO TABLE table1
    COLUMNS TERMINATED BY ","
    FORMAT AS CSV 
    (
    	SKIP_HEADER = 2
        TRIM_SPACE = TRUE
 	ENCLOSE = "'"
        ESCAPE = "\\"
    )
    (id, name, score)
)
WITH BROKER "csv_test"
(
    "username" = "hdfs_username",
    "password" = "hdfs_password"
)
PROPERTIES
(
    "timeout" = "3600"
);
```
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by chaoyli** [comment]
1. The columns name should come near the `COLUMNS`
```
    COLUMNS TERMINATED BY ","
    FORMAT AS CSV 
    (
        "trimspace"="true",
        "enclose"="'",
        "escape"="\\",
        "skipheader"="2"
    )
    (id, name, score)
```

```
    COLUMNS TERMINATED BY ","
    (id, name, score)
    FORMAT AS CSV 
    (
        "trimspace"="true",
        "enclose"="'",
        "escape"="\\",
        "skipheader"="2"
    )
```

**Comment by chaoyli** [comment]
2. How to write the `\t`, "\\t" or "\t"?

**Comment by zaorangyang** [comment]
> 1. The columns name should come near the `COLUMNS`
> 
> ```
>     COLUMNS TERMINATED BY ","
>     FORMAT AS CSV 
>     (
>         "trimspace"="true",
>         "enclose"="'",
>         "escape"="\\",
>         "skipheader"="2"
>     )
>     (id, name, score)
> ```
> 
> ```
>     COLUMNS TERMINATED BY ","
>     (id, name, score)
>     FORMAT AS CSV 
>     (
>         "trimspace"="true",
>         "enclose"="'",
>         "escape"="\\",
>         "skipheader"="2"
>     )
> ```

image
In fact, column list domain comes after format domain.

**Comment by zaorangyang** [comment]
> 

“\t”

**Comment by zaorangyang** [comment]
> 2. How to write the `\t`, "\t" or "\t"?
"\t"

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 36 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/BrokerFileGroup.java|20|20|100.00%|[]|

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 54 / 54 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/CsvFormat.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/BrokerFileGroup.java|20|20|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/load/BrokerFileGroup.java [hunk 1]
```diff
@@ -102,6 +103,17 @@ public class BrokerFileGroup implements Writable {
     // load from table
     private long srcTableId = -1;
     private boolean isLoadFromTable = false;
+    
+    // for csv
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/load/BrokerFileGroup.java:107]: why not store CsvFormat directly

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -2094,8 +2095,36 @@ private DataDescription getDataDescription(StarRocksParser.DataDescContext conte
...
+                StringLiteral stringLiteral = (StringLiteral) visit(formatPropsContext.escapeCharacter);
+                escape = stringLiteral.getValue();
+            }
+            String enclose = null;
+            if (formatPropsContext.encloseCharacter != null) {
+                StringLiteral stringLiteral = (StringLiteral) visit(formatPropsContext.encloseCharacter);
+                enclose = stringLiteral.getValue();
+            }
+            long skipheader = 0;
+            if (formatPropsContext.INTEGER_VALUE() != null) {
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2113]: what will happen when skipheader less than 0

**Comment by wangruin** [comment]
run starrocks_fe_unittest

**Comment by zaorangyang** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 47 / 48 (97.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/BrokerFileGroup.java|11|12|91.67%|[128]|
|:large_blue_circle:|com/starrocks/common/CsvFormat.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|4|4|100.00%|[]|

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_fe_unittest

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_fe_unittest

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 47 / 48 (97.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/BrokerFileGroup.java|11|12|91.67%|[128]|
|:large_blue_circle:|com/starrocks/common/CsvFormat.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DataDescription.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/load/BrokerFileGroup.java [hunk 3]
```diff
@@ -102,9 +103,18 @@ public class BrokerFileGroup implements Writable {
...
+    private CsvFormat csvFormat;
+
+    public static final String ESCAPE = "escape";
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/load/BrokerFileGroup.java:110]: Why not extract these constants into the file CSVFormat.java

File: fe/fe-core/src/main/java/com/starrocks/load/BrokerFileGroup.java [hunk 4]
```diff
@@ -102,9 +103,18 @@ public class BrokerFileGroup implements Writable {
...
+    public static final String ENCLOSE = "enclose";
+    public static final String TRIMSPACE = "trim_space";
+    public static final String SKIPHEADER = "skip_header";
 
     // for unit test and edit log persistence
     private BrokerFileGroup() {
+        this.csvFormat = new CsvFormat((byte) 0, (byte) 0, 0, false);
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/load/BrokerFileGroup.java:117]: Why not specify the default values in CsvFormat unifiedly?

**Pull request merged**